File: drivers/xmldocx/wrappers/ruby/XmlDocx.rb
# Creates a Word document out of a XML file.
#
# Example
#   test = XmlDocx.new("config.xml")
#   test.setDocumentProperties("settings.xml")
#   test.addContent("content.xml")
#   test.render

require 'pathname'

class XmlDocx
    def initialize(xml, debug = false)
        unless File.file?(xml)
            raise 'The file does not exist'
        end

        @config = Pathname.new(xml).realpath.to_s
        @content = Array.new
        @documentProperties = Array.new
        @debug = debug
        @xmldocxPath = File.dirname(__FILE__) + '/../'
    end

    # XML file with content to be added to the Word document.
    def addContent(xml)
        unless File.file?(xml)
            raise 'The file does not exist'
        end

        @content.push(Pathname.new(xml).realpath.to_s)
    end

    # XML file with the general properties of the Word document.
    def setDocumentProperties(xml)
        unless File.file?(xml)
            raise 'The file does not exist'
        end

        @documentProperties.push(Pathname.new(xml).realpath.to_s)
    end

    # Absolute path to the xmldocx folder
    def setXmlDocxPath(xmldocxPath)
        @xmldocxPath = xmldocxPath
    end

    # Renders Word document out of the XML data.
    def render
        system("php", @xmldocxPath + "xmldocx/XMLAPICommand.php", "-c" + @config, "-d" + @documentProperties * " ", "-b" + @content * " ")
    end
end

File: drivers/xmldocx/wrappers/node.js/XmlDocx.js
"use strict";

var fs = require("fs"),
    path = require("path"),
    exec = require("child_process").exec;

/**
 * Creates a Word document out of a XML file.
 *
 * Example:
 *  var test = new XmlDocx("config.xml");
 *  test.setDocumentProperties("settings.xml");
 *  test.addContent("content.xml");
 *  test.render();
 */
function XmlDocx(xml, debug) {
    if (!fs.existsSync(xml)) {
        throw "The file does not exist";
    }

    this.config = path.resolve(process.cwd(), xml);
    this.documentContent = [];
    this.documentProperties = [];
    this.debug = debug;
    this.xmldocxPath = __dirname + "/../";
}

/**
 * XML file with content to be added to the Word document.
 */
XmlDocx.prototype.addContent = function(xml) {
    if (!fs.existsSync(xml)) {
        throw "The file does not exist";
    }

    this.documentContent.push(path.resolve(process.cwd(), xml));
}

/**
 * XML file with the general properties of the Word document.
 */
XmlDocx.prototype.setDocumentProperties = function(xml) {
    if (!fs.existsSync(xml)) {
        throw "The file does not exist";
    }

    this.documentProperties.push(path.resolve(process.cwd(), xml));
}

/**
 * Absolute path to the xmldocx folder.
 */
XmlDocx.prototype.setXmldocxPath = function(xmldocxPath) {
    this.xmldocxPath = xmldocxPath;
}

/**
 * Renders Word document out of the XML data.
 */
XmlDocx.prototype.render = function() {
    var output = exec("php " + this.xmldocxPath + "xmldocx/XMLAPICommand.php -c" + this.config + " -d" + this.documentProperties.join(" ") + " -b'" + this.documentContent.join(" ") + "'");

    if (this.debug) {
        console.log(output);
    }
}

module.exports = XmlDocx;